flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/JobMasterPartitionTrackerImpl.java

```diff
@@ -78,6 +77,22 @@ public void startTrackingPartition(
                 producingTaskExecutorId, resultPartitionId, resultPartitionDeploymentDescriptor);
     }
 
+    @Override
+    public Collection<PartitionTrackerEntry<ResourceID, ResultPartitionDeploymentDescriptor>>
+            stopTrackingPartitionsFor(ResourceID producingTaskExecutorId) {
+        Preconditions.checkNotNull(producingTaskExecutorId);
+
+        // stop tracking partitions which are accommodated on TM
+        Set<ResultPartitionID> rIds =
+                partitionTable.getTrackedPartitions(producingTaskExecutorId).stream()
+                        .map(rId -> partitionInfos.get(rId).getMetaInfo())
+                        .filter(JobMasterPartitionTrackerImpl::isPartitionWithLocalResources)
+                        .map(desc -> desc.getShuffleDescriptor().getResultPartitionID())
+                        .collect(Collectors.toSet());
+        partitionTable.stopTrackingPartitions(producingTaskExecutorId);
```

Comment:
Looks to me this invocation is not needed.
Since we now have `partitionTable#getTrackedPartitions()`, maybe we can also fix `AbstractPartitionTracker#stopTrackingPartitionsFor(...)` which calls `partitionTable.stopTrackingPartitions(...)` twice.

Reply:
Sure I will remove this invocation ~

But I'm hesitate to think there's issue in `AbstractPartitionTracker#stopTrackingPartitionsFor` -- the idea is to stop tracking by tmID and get corresponding partitionID, then stop tracking by partitionID. Is there redundancy ? Did I get it wrong ?

Reply:
That's better ~